examples/bbox_annotator.py

```diff
@@ -0,0 +1,87 @@
+from magicgui.widgets import ComboBox, Container
+import napari
+import numpy as np
+
+
+# set up the annotation values and text display properties
+box_annotations = ['person', 'sky', 'camera']
+text_property = 'box_label'
+text_color = 'green'
+text_size = 30
+
+
+# create the GUI for selecting the values
+def create_label_menu(shapes_layer, label_property, labels):
+    """Create a label menu widget that can be added to the napari viewer dock
+
+    Parameters:
+    -----------
+    shapes_layer : napari.layers.Shapes
+        a napari shapes layer
+    label_property : str
+        the name of the shapes property to use the displayed text
+    labels : List[str]
+        list of the possible text labels values.
+
+    Returns:
+    --------
+    label_widget : magicgui.widgets.Container
+        the container widget with the label combobox
+    """
+    # Create the label selection menu
+    label_menu = ComboBox(label='text label', choices=labels)
+    label_widget = Container(widgets=[label_menu])
+
+    def update_label_menu(event):
+        """This is a callback function that updates the label menu when
+        the current properties of the Shapes layer change
+        """
+        new_label = str(shapes_layer.current_properties[label_property][0])
+        if new_label != label_menu.value:
+            label_menu.value = new_label
+
+    shapes_layer.events.current_properties.connect(update_label_menu)
+
+    def label_changed(event):
+        """This is acallback that update the current properties on the Shapes layer
+        when the label menu selection changes
+        """
+        selected_label = event.value
+        current_properties = shapes_layer.current_properties
+        current_properties[label_property] = np.asarray([selected_label])
+        shapes_layer.current_properties = current_properties
+
+    label_menu.changed.connect(label_changed)
+
+    return label_widget
+
+
+with napari.gui_qt():
+    # create a viewer with a fake t+2D image
+    viewer = napari.view_image(np.random.random((5, 200, 200)))
```

Comment:
Why not use the skimage cameraman here, especially given your box labels :-)

It's only 2D, but i think it's always nicer to look at a real image when possible rather than random pixels

Reply:
Good point! I can add that. I'll make stack of the camerman image with different slices translated/rotated.

Reply:
Fun!!